iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/connector/protocol/thrift/async/IoTDBThriftAsyncClientManager.java

```diff
@@ -179,7 +193,31 @@ public void onError(Exception e) {
     }
 
     if (exception.get() != null) {
-      throw new PipeConnectionException("Failed to handshake.", exception.get());
+      if (statusCode.get() == TSStatusCode.PIPE_TYPE_ERROR.getStatusCode()) {
```

Comment:
How to judge the diff between network error and cluster id error>?

Reply:
The status code of cluster id error is PIPE_TYPE_ERROR. Otherwise, it's network error.